src/test/scala/scala/collection/decorators/MapDecoratorTest.scala

```diff
@@ -74,4 +74,92 @@ class MapDecoratorTest {
 //    Assert.assertEquals(expected, zipped2)
   }
 
+  @Test
+  def mergingByKeyPerformsFullOuterJoin(): Unit = {
+    val arthur = "arthur.txt"
+
+    val tyson = "tyson.txt"
+
+    val sandra = "sandra.txt"
+
+    val allKeys = Set(arthur, tyson, sandra)
+
+    val sharedValue = 1
+
+    val ourChanges = Map(
+      (
+        arthur,
+        sharedValue
+      ),
+      (
+        tyson,
+        2
+      )
+    )
+
+    locally {
+      // In this test case, none of the associated values collide across keys...
+
+      val theirChanges = Map(
+        (
+          arthur,
+          sharedValue
+        ),
+        (
+          sandra,
+          3
+        )
+      )
+
+      Assert.assertEquals("Expect the same keys to appear in the join taken either way around.", ourChanges.mergeByKey(theirChanges).keySet, theirChanges
+        .mergeByKey(ourChanges)
+        .keys)
```

Comment:
This compares keySet and keys. Does it matter? Without a resolution to https://github.com/scala/scala/pull/10766 it's fruitless to speculate.

Reply:
Oops - yes, let me fix this…

Reply:
**DONE**